### ISSUE TITLE
ERROR: LVD not found, all data is after yearLast

### ISSUE THREAD

**Issue by mariuslam** [ESCOMP/CTSM#931]
### Brief summary of bug
I ran into this error when running FATES with hydro. I kept changing START_DATE, RUN_STARTDATE. But kept having this error :"ERROR: LVD not found, all data is after yearLast".

What I am trying to do is cycling three times over the 18years of forcing, to have 36 years of spinup the last 18 years for analysis.

### General bug information

release-clm5.0.30-5-g5220caaa
cime5.6.28-4-g1dc4e1c53
sci.1.33.0_api.8.1.0

**Does this bug cause significantly incorrect results in the model's science?** [No]

### Important details of your setup / configuration so we can reproduce the bug

        echo "____________________________________________Start changes"
	cd ${path4}/${GRIDNAME}_${name}
        ./xmlchange ATM_DOMAIN_PATH=$GENDOM_PATH,LND_DOMAIN_PATH=$GENDOM_PATH
        ./xmlchange ATM_DOMAIN_FILE=$ATMDOM,LND_DOMAIN_FILE=$ATMDOM
        ./xmlchange CLM_USRDAT_NAME=${GRIDNAME}_${prec}_${temp}
	./xmlchange DATM_MODE="CLM1PT_ERA_LAND"
	./xmlchange RESUBMIT=17,STOP_N=3,STOP_DATE=20190101
	./xmlchange STOP_OPTION=nyears
	./xmlchange NTASKS=1
	./xmlchange DIN_LOC_ROOT_CLMFORC="${path5}/ERAland"	
	./xmlchange JOB_WALLCLOCK_TIME="00:10:00"
	./xmlchange DATM_CLMNCEP_YR_ALIGN="2001" 
	./xmlchange DATM_CLMNCEP_YR_START="2001" 
	./xmlchange DATM_CLMNCEP_YR_END="2018" 
	./xmlchange DOUT_S="FALSE"
	./xmlchange GMAKE_J="8" 
	./xmlchange CLM_ACCELERATED_SPINUP="off"  
	cat > user_nl_clm << EOF
	cat> user_nl_datm <<EOF
&shr_strdata_nml
   vectors        = 'null'
   mapmask        = 'nomask','nomask','nomask','nomask','nomask'
   mapalgo        = 'nn','nn','nn','nn','nn'
   tintalgo       = 'coszen','nearest','linear','linear','lower'
   taxmode        = 'cycle','cycle','cycle','extend','extend'
   streams        = 'datm.streams.txt.CLM1PT_ERA_LAND.CLM_USRDAT.solar 2001 2001 2018',
                    'datm.streams.txt.CLM1PT_ERA_LAND.CLM_USRDAT.prec 2001 2001 2018',
                    'datm.streams.txt.CLM1PT_ERA_LAND.CLM_USRDAT.rest 2001 2001 2018',
                    'datm.streams.txt.presaero.clim_2000 1 1 1',
                    'datm.streams.txt.topo.observed 1 1 1'
/
EOF

### Important output or errors that show the problem
CESM_LOG:
 (t_initf) Read in prof_inparm namelist from: drv_in
 (t_initf) Using profile_disable=          F
 (t_initf)       profile_timer=                      4
 (t_initf)       profile_depth_limit=                4
 (t_initf)       profile_detail_limit=               2
 (t_initf)       profile_barrier=          F
 (t_initf)       profile_outpe_num=                  1
 (t_initf)       profile_outpe_stride=               0
 (t_initf)       profile_single_file=      F
 (t_initf)       profile_global_stats=     T
 (t_initf)       profile_ovhd_measurement= F
 (t_initf)       profile_add_detail=       F
 (t_initf)       profile_papi_enable=      F
 ERROR: (shr_stream_findBounds) ERROR: LVD not found, all data is after yearLast
Image              PC                Routine            Line        Source
cesm.exe           00000000011310F6  Unknown               Unknown  Unknown
cesm.exe           0000000000DCA7AE  shr_abort_mod_mp_         114  shr_abort_mod.F90
cesm.exe           0000000000F2A998  shr_stream_mod_mp        3039  shr_stream_mod.F90
cesm.exe           0000000000F1E52A  shr_stream_mod_mp        1148  shr_stream_mod.F90
cesm.exe           0000000000E6D717  shr_dmodel_mod_mp         622  shr_dmodel_mod.F90
cesm.exe           0000000000F07196  shr_strdata_mod_m         743  shr_strdata_mod.F90
cesm.exe           00000000004E0D5C  datm_comp_mod_mp_         664  datm_comp_mod.F90
cesm.exe           00000000004E6E6F  datm_comp_mod_mp_         551  datm_comp_mod.F90
cesm.exe           00000000004DF639  atm_comp_mct_mp_a         167  atm_comp_mct.F90
cesm.exe           0000000000433DC6  component_mod_mp_         267  component_mod.F90
cesm.exe           0000000000422CEB  cime_comp_mod_mp_        1231  cime_comp_mod.F90
cesm.exe           0000000000430F5E  MAIN__                    114  cime_driver.F90
cesm.exe           000000000041458E  Unknown               Unknown  Unknown
libc-2.17.so       00002AF5B78FA495  __libc_start_main     Unknown  Unknown
cesm.exe           00000000004144A9  Unknown               Unknown  Unknown
application called MPI_Abort(MPI_COMM_WORLD, 1001) - process 0

ATM_LOG:
(datm_comp_init)  flds_strm = strm_swdn:strm_precn:strm_tbot:strm_wind:strm_pbot:strm_tdew:strm_lwdn
(shr_dmodel_readstrm) open   : /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc
(shr_dmodel_readstrm) file lb: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       1
(shr_dmodel_readstrm) file ub: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       2
(shr_dmodel_readstrm) open   : /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc
(shr_dmodel_readstrm) file lb: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       1
(shr_dmodel_readstrm) file ub: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       2
(shr_dmodel_readstrm) open   : /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc
(shr_dmodel_readstrm) file lb: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       1
(shr_dmodel_readstrm) file ub: /cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc       2
(shr_stream_findBounds) ERROR: LVD not found, all data is after yearLast
 ERROR: (shr_stream_findBounds) ERROR: LVD not found, all data is after yearLast

**Comment by ekluzek** [comment]
LVD (Least Valid Date) means all the data in the file is beyond the year it's trying to find. Your streams are setup to expect years 2001 to 2018 on your forcing files. What is time on the first one of those files: 

/cluster/projects/nn9373k/marlam/inputdata_CLM1PT_pan/ERAland/1x1_Merrelva_era_P1_T3/2001-01.nc

And what are the attributes for the time field on that file as well (should be something like days since a given date).

**Comment by mariuslam** [comment]
This is in the file:
days since 2001-01-01 00:00:00
time = UNLIMITED ; // (248 currently)

**Comment by mariuslam** [comment]
time = 0, 0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875, 1, 1.125, 1.25, 
    1.375, 1.5, 1.625, 1.75, 1.875, 2, 2.125, 2.25, 2.375, 2.5, 2.625, 2.75, 
    2.875, 3, 3.125, 3.25, 3.375, 3.5, 3.625, 3.75, 3.875, 4, 4.125, 4.25, 
    4.375, 4.5, 4.625, 4.75, 4.875, 5, 5.125, 5.25, 5.375, 5.5, 5.625, 5.75, 
    5.875, 6, 6.125, 6.25, 6.375, 6.5, 6.625, 6.75, 6.875, 7, 7.125, 7.25, 
    7.375, 7.5, 7.625, 7.75, 7.875, 8, 8.125, 8.25, 8.375, 8.5, 8.625, 8.75, 
    8.875, 9, 9.125, 9.25, 9.375, 9.5, 9.625, 9.75, 9.875, 10, 10.125, 10.25, 
    10.375, 10.5, 10.625, 10.75, 10.875, 11, 11.125, 11.25, 11.375, 11.5, 
    11.625, 11.75, 11.875, 12, 12.125, 12.25, 12.375, 12.5, 12.625, 12.75, 
    12.875, 13, 13.125, 13.25, 13.375, 13.5, 13.625, 13.75, 13.875, 14, 
    14.125, 14.25, 14.375, 14.5, 14.625, 14.75, 14.875, 15, 15.125, 15.25, 
    15.375, 15.5, 15.625, 15.75, 15.875, 16, 16.125, 16.25, 16.375, 16.5, 
    16.625, 16.75, 16.875, 17, 17.125, 17.25, 17.375, 17.5, 17.625, 17.75, 
    17.875, 18, 18.125, 18.25, 18.375, 18.5, 18.625, 18.75, 18.875, 19, 
    19.125, 19.25, 19.375, 19.5, 19.625, 19.75, 19.875, 20, 20.125, 20.25, 
    20.375, 20.5, 20.625, 20.75, 20.875, 21, 21.125, 21.25, 21.375, 21.5, 
    21.625, 21.75, 21.875, 22, 22.125, 22.25, 22.375, 22.5, 22.625, 22.75, 
    22.875, 23, 23.125, 23.25, 23.375, 23.5, 23.625, 23.75, 23.875, 24, 
    24.125, 24.25, 24.375, 24.5, 24.625, 24.75, 24.875, 25, 25.125, 25.25, 
    25.375, 25.5, 25.625, 25.75, 25.875, 26, 26.125, 26.25, 26.375, 26.5, 
    26.625, 26.75, 26.875, 27, 27.125, 27.25, 27.375, 27.5, 27.625, 27.75, 
    27.875, 28, 28.125, 28.25, 28.375, 28.5, 28.625, 28.75, 28.875, 29, 
    29.125, 29.25, 29.375, 29.5, 29.625, 29.75, 29.875, 30, 30.125, 30.25, 
    30.375, 30.5, 30.625, 30.75, 30.875 ;
}

**Comment by ekluzek** [comment]
OK, that all looks good. What is RUN_STARTDATE? Do an ./xmlquery to show it's value...

And your datafiles do go until the end of 2018 also right? There needs to be no gaps in the forcing data. So you should have Jan/1/2001 until Dec/31/2018.

**Comment by ekluzek** [comment]
I tried a test case from the latest fates_next_api branch that sets up for single-point forcing, and it is working:

SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpGs.hobart_intel.clm-USUMB

That's with: release-clm5.0.30-5-g5220caaa
and: sci.1.33.0_api.8.1.0

Here's the list of setting changes for that case:
cat shell_commands 
# shell commands to execute xmlchange --force commands written by PTCLMmkdata:
# ./PTCLMmkdata --cesm_root ../../../.. -s US-UMB -d /glade/p/cesm/cseg/inputdata --mydatadir=/glade/p/cesm/cseg/inputdata/lnd/clm2/PTCLMmydatafiles.c171024 
./xmlchange --force CLM_USRDAT_NAME=1x1pt_US-UMB
./xmlchange --force DATM_CLMNCEP_YR_START=1999
./xmlchange --force DATM_CLMNCEP_YR_END=2006
# Comment this out if NINST_LND is greater than 1 (see: http://bugs.cgd.ucar.edu/show_bug.cgi?id=2521)
./xmlchange --force MPILIB=mpi-serial
./xmlchange --force ATM_DOMAIN_PATH='$DIN_LOC_ROOT/lnd/clm2/PTCLMmydatafiles.c171024/1x1pt_US-UMB'
./xmlchange --force LND_DOMAIN_PATH='$DIN_LOC_ROOT/lnd/clm2/PTCLMmydatafiles.c171024/1x1pt_US-UMB'
./xmlchange --force ATM_DOMAIN_FILE=domain.lnd.1x1pt_US-UMB_navy.171024.nc
./xmlchange --force LND_DOMAIN_FILE=domain.lnd.1x1pt_US-UMB_navy.171024.nc
./xmlchange --force --append CLM_BLDNML_OPTS='-mask navy -no-crop'
./xmlchange --force CALENDAR=GREGORIAN
./xmlchange --force DOUT_S=FALSE
./xmlchange --force ATM_NCPL=24
./xmlchange --force RUN_STARTDATE=1999-01-01
./xmlchange --force DATM_CLMNCEP_YR_ALIGN=1999
./xmlchange --force DIN_LOC_ROOT=/fs/cgd/csm/inputdata
./xmlchange --force DIN_LOC_ROOT_CLMFORC='$DIN_LOC_ROOT/lnd/clm2/PTCLMmydatafiles.c171024'
[erik@hobart SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpGs.hobart_intel.clm-USUMB.20200226_161203_l8tn93]$ cat u
user_nl_clm     user_nl_cpl     user_nl_datm    user_nl_mosart  
[erik@hobart SMS_D_Lm1_Mmpi-serial.CLM_USRDAT.I1PtClm50SpGs.hobart_intel.clm-USUMB.20200226_161203_l8tn93]$ cat user_nl_clm 
! user_nl_clm namelist options written by PTCLMmkdata:
! ./PTCLMmkdata --cesm_root ../../../.. -s US-UMB -d /glade/p/cesm/cseg/inputdata --mydatadir=/glade/p/cesm/cseg/inputdata/lnd/clm2/PTCLMmydatafiles.c171024 
 fsurdat = '$DIN_LOC_ROOT/lnd/clm2/PTCLMmydatafiles.c171024/1x1pt_US-UMB/surfdata_1x1pt_US-UMB_16pfts_Irrig_CMIP6_simyr2000_c171024.nc'
 hist_nhtfrq = 0
 hist_mfilt  = 1200

I don't see important configuration differences between that case and the one that isn't working for @mariuslam.

**Comment by ekluzek** [comment]
@mariuslam and I met and I went over his cases. He has a case essentially identical to the one that fails that works based off of ctsm1.0.dev083. The one that fails is based off of fates_next_api as I gave above. He had cases working based on an earlier version of fates_next_api. So the problem seems to be in the cime version being used in the two cases. The datm files are identical between the case on ctsm master that works and the one on fates_next_api that fails. So I can't explain why this is happening.

**Comment by ekluzek** [comment]
Jackie also tried a case and her case worked as well. So the problem is looking like it's specific to @mariuslam setup. But, I haven't seen anything wrong there, so I'm not sure what.

We did suggest to try using a slightly older version of fates_next_api (since an older version was working). That version also has an older version of cime, and his changes to cime will need to be moved over as well.

**Comment by ekluzek** [comment]
We solved the problem.

We thought this problem had to do with @mariuslam new files, so that's what we examined intently. The problem was actually in the prescribed aerosol stream though. In previous versions the file being used had year represented as year "1" for a single year being used (for 2000_control). The new setup had years from 1849 and onward for the entire historical period. The previous changes to user_nl_datm that were being used had the start and stop year for presaero as "1" (for the previous file), the new setup needed it to be "2000". Once @mariuslam put that in the model is now working for him.

The take away is that sometimes user_* changes from a previous case may NOT work in a new case when the model has been updated. I'm not sure there's a way to prevent that. But, we should realize that user_* files have some of the same fragility as SourceMods for model updates. It might even be a good idea to put a disclaimer in the user_nl_clm file to that effect. And for completeness probably other user_nl_* files should have that same warning as well.

But, since this is working, I'm closing this.
